### PR TITLE
Include two different stages for building TGI image:

### PR DESCRIPTION
**Pull request by mfuntowicz**
- Default standalone image
- Inference Endpoint specific image

### CODE REVIEW

File: text-generation-inference/docker/Dockerfile [hunk 1]
```diff
@@ -119,10 +119,16 @@ COPY --from=builder /usr/src/target/release/text-generation-launcher /usr/local/
...
 RUN chmod +x entrypoint.sh
 
-ENTRYPOINT ["./entrypoint.sh"]
\ No newline at end of file
+ENTRYPOINT ["./entrypoint.sh"]
+
+# TPU compatible image
+FROM tpu_base as final_image
+ENTRYPOINT text-generation-launcher
+CMD ["--json-output"]
```
Comment (tengomucho) [inline text-generation-inference/docker/Dockerfile:134]: this seems to revert https://github.com/huggingface/optimum-tpu/pull/28/commits/8756c8a1753eb538b9069dc88b1fe001403ef68d, are you sure about it?

**Review by tengomucho — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -43,12 +43,20 @@ clean:
...
 
 tpu-tgi:
-	docker build --rm -f text-generation-inference/Dockerfile \
+	docker build --rm -f text-generation-inference/docker/Dockerfile \
 	             --build-arg VERSION=$(VERSION) \
 	             --build-arg TGI_VERSION=$(TGI_VERSION) \
 				 -t huggingface/optimum-tpu:$(VERSION)-tgi .
 	docker tag huggingface/optimum-tpu:$(VERSION)-tgi huggingface/optimum-tpu:latest
 
+tpu-tgi-ie:
```
Comment (tengomucho) [inline Makefile:52]: target `tgi_docker_test` should depend on this (and we should probably fix it at some point)

File: text-generation-inference/docker/Dockerfile (hunk 1 above)
Comment (mfuntowicz) [reply text-generation-inference/docker/Dockerfile:134]: For the case of non inference-endpoints (IE) Docker image we want the image to provide their own argument, so this was correct and the changes we did were too focus on IE loosing flexibility in other deployment scenarios

File: Makefile (hunk 2 above)
Comment (mfuntowicz) [reply Makefile:52]: Does the `tgi_docker_test` needs to be run on an inference-endpoints specific docker image?

File: text-generation-inference/docker/Dockerfile (hunk 1 above)
Comment (tengomucho) [reply text-generation-inference/docker/Dockerfile:135]: Ok, in that case we might want to keep 2 docker images, one for TGI non inference-endpoints (IE), the other for IE. Or, if we are not interested in maintaining one of those, we should probably drop the image and the associated targets.

**Review by tengomucho — APPROVED** [review]

File: text-generation-inference/docker/Dockerfile (hunk 1 above)
Comment (mfuntowicz) [reply text-generation-inference/docker/Dockerfile:135]: Yes but we are definitely interested in keeping both and potentially adding a third one for vertex if it makes sense later 👍🏻

**Review by tengomucho — APPROVED** [review]

**Pull request merged**